docs/source/java/examples/quickstartguide.rst

```diff
@@ -0,0 +1,277 @@
+
+VectorSchemaRoot is somewhat analogous to tables and record batches in the other
+Arrow implementations.
+
+**VectorSchemaRoot**: Create a dataset with metadata that contains integer age and
+string names of data.
+
+.. code-block:: Java
+
+    import org.apache.arrow.memory.BufferAllocator;
+    import org.apache.arrow.memory.RootAllocator;
+    import org.apache.arrow.vector.IntVector;
+    import org.apache.arrow.vector.VarCharVector;
+    import org.apache.arrow.vector.VectorSchemaRoot;
+    import org.apache.arrow.vector.types.pojo.ArrowType;
+    import org.apache.arrow.vector.types.pojo.Field;
+    import org.apache.arrow.vector.types.pojo.FieldType;
+    import org.apache.arrow.vector.types.pojo.Schema;
+
+    import java.nio.charset.StandardCharsets;
+    import java.util.HashMap;
+    import java.util.Map;
+    import static java.util.Arrays.asList;
+
+    Map<String, String> metadataField = new HashMap<>();
+    metadataField.put("K1-Field", "K1F1");
+    metadataField.put("K2-Field", "K2F2");
+    Field a = new Field("Column-A-Age", FieldType.nullable(new ArrowType.Int(32, true)), null);
+    Field b = new Field("Column-B-Name", new FieldType(true, new ArrowType.Utf8(), /*dictionary*/ null, metadataField), null);
+    Map<String, String> metadataSchema = new HashMap<>();
+    metadataSchema.put("K1-Schema", "K1S1");
+    metadataSchema.put("K2-Schema", "K2S2");
+    Schema schema = new Schema(asList(a, b), metadataSchema);
+    System.out.println("Field A: " + a);
+    System.out.println("Field B: " + b + ", Metadata: " + b.getMetadata());
+    System.out.println("Schema: " + schema);
+    try(BufferAllocator rootAllocator = new RootAllocator();
+        VectorSchemaRoot vectorSchemaRoot = VectorSchemaRoot.create(schema, rootAllocator)){
+        vectorSchemaRoot.setRowCount(3);
+        try(IntVector intVectorA = (IntVector) vectorSchemaRoot.getVector("Column-A-Age");
+            VarCharVector varCharVectorB = (VarCharVector) vectorSchemaRoot.getVector("Column-B-Name")) {
+            intVectorA.allocateNew(3);
+            intVectorA.set(0, 10);
+            intVectorA.set(1, 20);
+            intVectorA.set(2, 30);
+
+            varCharVectorB.allocateNew(3);
+            varCharVectorB.set(0, "Dave".getBytes(StandardCharsets.UTF_8));
+            varCharVectorB.set(1, "Peter".getBytes(StandardCharsets.UTF_8));
+            varCharVectorB.set(2, "Mary".getBytes(StandardCharsets.UTF_8));
+
+            System.out.println("VectorSchemaRoot: \n" + vectorSchemaRoot.contentToTSVString());
+        }
+    }
+
+Interprocess Communication (IPC)
+********************************
+
+Arrow data can be written to and read from disk, and both of these can be done in
+a streaming and/or random-access fashion depending on application requirements.
+
+**Create a IPC File or Random Access Format**
+
+Write File or Random Access Format: Write to a file a dataset with metadata
+that contains integer age and string names of data.
```

Comment:
"Example: write the dataset from the previous example to an Arrow random-access file:"

Reply:
Changed